### ISSUE TITLE
Unwanted Tidal MQA playback with audio dropouts

### ISSUE THREAD

**Issue by SirJohnDoe** [music-assistant/support#2911]
### What version of Music Assistant has the issue?

2.2.3

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

MA plays back MQA versions of Tidal albums. The MQA versions don't seem to work very well, because I hear audio dropouts.

The exact same albums in the Tidal app play as FLAC with Tidal Connect and don't have these issues.

Seeing that Tidal stopped supporting MQA (not even available with Tidal Connect anymore) it would be better if MA would always revert to the best FLAC version available, and not play MQA albums.

### How to reproduce

Add this album to favourites and sync to MA: https://tidal.com/album/4488861?u

It happens both with HIGH_LOSSLESS profile as well as HIGH_RES profiles

### Music Providers

Tidal

### Player Providers

UPnP

### What version of Home Assistant Core are your running

2024.9.1

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Linux

**Comment by marcelveldt** [comment]
I'm pretty sure the MQA tracks are not even provided anymore in the catalog so this is a bit surprising.
I'll try to reproduce

**Comment by SirJohnDoe** [comment]
Not sure what changed in 2.2.4, but now the album continuously switches between (24 bit) FLAC and MQA.

https://github.com/user-attachments/assets/26b3a660-8a16-4a42-b223-2feda283f59b

Also happens on another album, and also happens when removing, restarting and adding Tidal again.

**Comment by marcelveldt** [comment]
I have no idea what is going on but I cant reproduce any of this, not do I get an MQA stream from Tidal.
Did you try this on another player ? DLNA is famous for its flaky-ness.

Maybe also go a bit through the player settings for this player and try setting the HTTP Profile to Chunked.

**Comment by OzGav** [comment]
@jozefKruszynski thoughts?

**Comment by jozefKruszynski** [comment]
As @marcelveldt said this shouldn't be possible as the content should simply no longer be available on tidal.
Also switching stream type during playback is not something that should be possible.

@SirJohnDoe Did you completely remove and re-add tidal after the auth change?
The change to the pkce login type requires the complete removal before being re-added

**Comment by SirJohnDoe** [comment]
A full reinstall of 2.2.5, but also latest dev, solved it. I only get FLAC now. Perhaps there was some remnant that linked to MQA albums or something.

**Comment by OzGav** [comment]
@jozefKruszynski

**Comment by jozefKruszynski** [comment]
I have no way of reproducing this as I have nothing that is capable of playing mqa.
Only thing I can think is that the player firmware is booked as Tidal do not have any mqa content anymore.

**Comment by SirJohnDoe** [comment]
> I have no way of reproducing this as I have nothing that is capable of playing mqa. Only thing I can think is that the player firmware is booked as Tidal do not have any mqa content anymore.

If I start the exact same album through Tidal (Connect) on the amp it never switches to MQA. It appears the MQA is still available in the API that MA uses, or cached somehow? When I first saved these tracks they were possibly still in MQA format. But I've done multiple full reinstalls of MA in the last few months and always did a full resync with Tidal. 

Does MA tell my player that it should start a Tidal track, like a chromecast where the player makes a direct connection with the streaming provider? If that's the case, and MA is not serving the audio, I can imagine caching can be the issue on my player.

**Comment by jozefKruszynski** [comment]
> > I have no way of reproducing this as I have nothing that is capable of playing mqa. Only thing I can think is that the player firmware is booked as Tidal do not have any mqa content anymore.
> 
> If I start the exact same album through Tidal (Connect) on the amp it never switches to MQA. It appears the MQA is still available in the API that MA uses, or cached somehow? When I first saved these tracks they were possibly still in MQA format. But I've done multiple full reinstalls of MA in the last few months and always did a full resync with Tidal.
> 
> Does MA tell my player that it should start a Tidal track, like a chromecast where the player makes a direct connection with the streaming provider? If that's the case, and MA is not serving the audio, I can imagine caching can be the issue on my player.

No, we grab the stream directly. It could well be the unofficial api that we use that may cause the issue.
I think there was recently an update there so I'll see if there's anything interesting happening over that way.

**Comment by OzGav** [comment]
@jozefKruszynski Any more needed to be done here?

**Comment by jozefKruszynski** [comment]
Let's keep it open until after the tidalapi 0.8.2 merge to see if that will help.

**Comment by OzGav** [comment]
@jozefKruszynski any update?

**Comment by jozefKruszynski** [comment]
Well we now have tidalapi 0.8.3, we would need an update from @SirJohnDoe as to whether the issue is resolved on their end.

**Comment by OzGav** [comment]
@SirJohnDoe please advise if this is still an issue. This will be closed soon if there is no follow up.

**Comment by SirJohnDoe** [comment]
Still occurs with 2.4.0b12. After installing it worked without issue, I think due to Volume normalisation being turned on by default for UPnP devices. After turning it off (the volume fluctuaties every few seconds, so it's not usable) the issue with MQA happens again.

**Comment by OzGav** [comment]
@jozefKruszynski thoughts?

**Comment by jozefKruszynski** [comment]
I have no clue whatsoever sorry.

**Comment by OzGav** [comment]
@SirJohnDoe In your video what display is showing the switch to MQA? Given that Tidal has removed all MQA tracks why is that not just a spurious indication?

**Comment by SirJohnDoe** [comment]
I ended up resetting my player, this had no effect. But I just installed the latest beta, 2.4.0b26, and the issue seems to be resolved 😊. I noticed that when I searched for a problematic track there was an error saying "track://79530261 not found on provider tidal" (after a fresh install of the latest beta), which makes me think that somehow the old MQA tracks were still retrieved and played somehow in the previous versions.

@OzGav my evo 150 player, and everytime it switched the audio dropped out, so it wasn't just on the display. Seeing that Tidal Connect worked flawlessly, and a reset of my player didn't work, it seems that the MQA was still send somehow. It didn't happen with other player providers, but those also never supported MQA as far as I know.
